{
  final ObjectMember.AuthorizationException authorizationException=causalChainOf(ex,ObjectMember.AuthorizationException.class);
  if (authorizationException != null) {
    this.authorizationCause=true;
    this.mainMessage=authorizationException.getMessage();
  }
 else {
    this.authorizationCause=false;
    if (recognizedMessageIfAny != null) {
      this.recognized=true;
      this.mainMessage=recognizedMessageIfAny;
    }
 else {
      this.recognized=false;
      Iterable<NonRecoverableException> appEx=Iterables.filter(Throwables.getCausalChain(ex),NonRecoverableException.class);
      Iterator<NonRecoverableException> iterator=appEx.iterator();
      NonRecoverableException nonRecoverableException=iterator.hasNext() ? iterator.next() : null;
      this.mainMessage=nonRecoverableException != null ? nonRecoverableException.getMessage() : MAIN_MESSAGE_IF_NOT_RECOGNIZED;
    }
  }
  stackTraceDetailList=asStackTrace(ex);
}
